### DISCUSSION TITLE
Multi Tenancy ,how to flow?

### DISCUSSION THREAD

**Discussion by sc-hero** [filamentphp/filament#8140] [General]
I have announcement module , If I will post an announcement post, notification sent! and In filament multi tenancy, I attached announcement with foreign key team_id.So, If team_id == 1, One team only can see announcement , this is right, but, As the requirement, I want to show announcement to multiple team, so foreign team_id - [1,2,3..], This logic is right, Let me know multi tenancy flow.Please can you explain?

**Comment by micraux** [comment]
If I understand correctly, you would like 1 or many `announcements` being seen by 1 or many `teams`.
So, first you should create a pivot table as you have a many-to-many relationship between `announcements` and `teams`.
So far, I don't know if the multi tenancy feature works with a many-to-many relationship. Haven't tried yet...

**Reply by danharrin** [reply]
It does

**Reply by micraux** [reply]
Great!

**Comment by sc-hero** [comment]
Is it compatible filament 3 and stancl/tenancy packate without configuration filament builtin ternancy?

**Reply by micraux** [reply]
Yes, you can do that. But you will need to "get your hands dirty" a little bit...

In fact, I tried myself 2 months ago when I discovered Filament (v2) and wanted to use it for a non-profit project.
Then when Filament v3 came (with the multi tenant feature), I tested it and it was perfect for me so I kept the "filament-only" solution.

I prefer to keep the minimum number of packages (as far as I can): I think it is better in order to avoid incompatibilities when new versions come.
